packages/atomic/src/components/commerce/atomic-product-text/atomic-product-text.spec.ts

```diff
@@ -0,0 +1,534 @@
+        });
+
+        expect(element.shouldHighlight).toBe(true);
+      });
+
+      it('should support highlighting for #excerpt field', async () => {
+        const element = await renderComponent({
+          field: 'excerpt',
+          shouldHighlight: true,
+        });
+
+        expect(element.shouldHighlight).toBe(true);
+      });
+
+      it('should not support highlighting for other fields', async () => {
+        const element = await renderComponent({
+          field: 'ec_brand',
+          shouldHighlight: true,
+        });
+
+        const commerceText = locators.getCommerceText(element);
+        expect(commerceText).toBeDefined();
+      });
+    });
+
+    describe('error handling in highlighting', () => {
+      it('should handle errors gracefully when highlighting fails', async () => {
+        const productWithHighlights = buildFakeProduct({
+          ec_name: 'Test Product Name',
+          nameHighlights: [
+            {
+              offset: 5,
+              length: 7,
+            },
+          ],
+        });
+
+        const element = await renderComponent({
+          field: 'ec_name',
+          shouldHighlight: true,
+          product: productWithHighlights,
+        });
+
+        expect(element).toBeDefined();
+        expect(element.shadowRoot).toBeDefined();
+      });
+    });
+  });
+
+  describe('edge cases', () => {
+    it('should display nothing for numeric #field values', async () => {
+      const productWithNumericField = buildFakeProduct({
+        additionalFields: {
+          numeric_field: 12345,
+        },
+      });
+      const element = await renderComponent({
+        field: 'numeric_field',
+        product: productWithNumericField,
+      });
+      const commerceText = locators.getCommerceText(element);
+      expect(commerceText?.shadowRoot?.textContent).toBe('');
+    });
+
+    it('should display nothing for boolean #field values', async () => {
+      const productWithBooleanField = buildFakeProduct({
+        additionalFields: {
+          boolean_field: true,
+        },
+      });
+      const element = await renderComponent({
+        field: 'boolean_field',
+        product: productWithBooleanField,
+      });
+      const commerceText = locators.getCommerceText(element);
+      expect(commerceText?.shadowRoot?.textContent).toBe('');
+    });
+
+    it('should handle special characters in #field values', async () => {
+      const productWithSpecialChars = buildFakeProduct({
+        additionalFields: {
+          special_field: 'Value with éàü & symbols!',
+        },
+      });
+
+      const element = await renderComponent({
+        field: 'special_field',
+        product: productWithSpecialChars,
+      });
+
+      const commerceText = locators.getCommerceText(element);
+      expect(commerceText?.shadowRoot?.textContent).toContain(
+        'Value with éàü & symbols!'
+      );
+    });
+  });
+
+  describe('different interface types', () => {
```

Comment:
In general I'm not a huge fan of these describes; they kind of break the English in the test reports...

At the moment, the established guidelines for describe blocks are:

- wrap all tests in a file under a single describe block (ClassName / atomic-component-name / util-file-name)
- use describe blocks for conditions that include multiple tests (when ...)
- use describe blocks for testing methods within a class (#methodName)